src/corva/models/scheduled.py

```diff
@@ -48,6 +54,23 @@ class ScheduledEvent(BaseEvent):
         None, description='Day shift start time'
     )
 
+    @pydantic.validator('schedule_start', 'schedule_end')
+    def set_timestamp(cls, v) -> int:
+        """Casts Unix timestamp from milliseconds to seconds.
+
+        Casts Unix timestamp from millisecond to seconds, if provided timestamp is in
+          milliseconds.
+        """
+
+        try:
+            # the function will raise, if provided timestamp is in milliseconds,
+            # because the year will always be too big to fit into the datetime instance
+            datetime.utcfromtimestamp(v)
+        except ValueError:
+            v /= 1000  # cast value from milliseconds to seconds
```

Comment:
I think you can cast the value to an int here only, right?

Reply:
if value doesn't need cast from ms to sec and comes as float, it will be returned as float
so we make sure that return type is int, by casting it before returning

Reply:
> I think you can cast the value to an int here only, right?

that is right. as our validation occurs after pydantic one and validator param `always=False`: we always get int as input. so if cast from ms to sec not needed, we can just return the value